internal/audit/agent_pool.go

```diff
@@ -0,0 +1,84 @@
+// Copyright 2025 Mia srl
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package audit
+
+import (
+	"slices"
+
+	"github.com/rond-authz/rond/logging"
+	"github.com/rond-authz/rond/types"
+)
+
+const (
+	AgentStorageLog     = "log"
+	AgentStorageMongoDB = "mongodb"
+)
+
+type MongoAgentPoolOptions struct {
+	Client         types.MongoClient
+	CollectionName string
+}
+
+type AgentPoolOptions struct {
+	Logger         logging.Logger
+	Labels         Labels
+	Storages       []string
+	MongoDBStorage MongoAgentPoolOptions
+}
+
+func NewAgentPool(options AgentPoolOptions) AgentPool {
+	return &agentPool{
+		options: options,
+	}
+}
+
+type agentPool struct {
+	options AgentPoolOptions
+}
+
+func (c *agentPool) New() Agent {
+	agents := make([]Agent, 0)
```

Comment:
This should be initialized after the return of noopAgent

Reply:
done